### ISSUE TITLE
FID

### ISSUE THREAD

**Issue by kongcheng-996** [HKU-MedAI/GEM-3D#6]
Thanks for the method you proposed and for open-sourcing the code, but I ran into some issues reproducing the FID metric from the paper。

I used the provided pre-trained weights for generation, and then extracted slices from the three planes of the generated full CT for FID calculation across the three planes. I also used the feature extraction model with network = torch.hub.load("Warvito/radimagenet-models", model="radimagenet_resnet50", verbose=True). But I couldn't get the FID metrics reported in the paper. The FID I calculated was even very high. The calculated fid is even over 100. I'm kind of not sure where the problem happened when calculating FID.

If possible, could you provide the FID calculation code or more details on how FID is calculated, as well as the criteria for selecting slices? Thanks.

**Comment by Advocate99** [comment]
Hi, could you post your calculation code so that I can help to find some cues?
The code I used was a normal one for FID and I do not have it now (let me try to find it later, this may be okay: https://github.com/Project-MONAI/MONAI/blob/dev/monai/metrics/fid.py).

**Comment by kongcheng-996** [comment]
Thanks for your reply. After referring to the code you provided, I tested the new code, but the FID scores for the three sections are still quite high: 1054.392794, 1436.296181, and 968.978316. 
I’ve provided my FID calculation code. Thanks again for your reply.

[test_fid2.py](https://github.com/user-attachments/files/29540941/test_fid2.py)

**Comment by Advocate99** [comment]
Hi. Do the generated images look normal? 1000 fid means something wrong.
I think the major reason is that the data processing is not as the way the pretrained model expect. I cannot remember which model repo I used for FID but I try to find the code that also used Warvito/radimagenet-models as shown in your code. The code (https://github.com/Warvito/generative_brain_controlnet/blob/main/src/python/testing/compute_fid.py) did not used std for normalization and you can check it more carefully. You can also try to check some other details like whether it required per-slice normalization as I notice that per-volume normalization is used and the function preprocess_slice_batch does not operate the image slices that FID calculation requires.

**Comment by kongcheng-996** [comment]
Thanks for your reply. Through your link, I found the reason for my mistake. When processing the data, I subtracted the mean and then divided by the std. But actually, I only needed to subtract the mean. Now the FID data I got is similar to the data in the paper.
Thanks again for your reply and such excellent work.

**Comment by Advocate99** [comment]
Glad to hear that!
